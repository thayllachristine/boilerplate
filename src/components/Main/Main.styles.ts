import styled, { keyframes } from 'styled-components';

const Wrapper = styled.main`
  align-items: center;
  background-color: #333333;
  color: #fff;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  padding: 3rem;
  width: 100%;
`;

const Title = styled.h1`
  font-size: 2.5rem;
  margin-bottom: 2rem;
`;

const Subtitle = styled.h2`
  font-size: 2rem;
  font-weight: 400;
  margin-bottom: 3rem;
`;

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Image = styled.img`
  animation: ${rotate} 10s linear infinite;
  height: min(30rem, 100%);
  width: min(30rem, 100%);
`;

export { Wrapper, Title, Subtitle, Image };
