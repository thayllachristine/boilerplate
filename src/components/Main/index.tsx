import * as S from './Main.styles';

const Main = () => (
  <S.Wrapper>
    <S.Title>boilerplate</S.Title>
    <S.Subtitle data-testid="subtitle">
      A simple project starter to work with React, NextJS, TypeScript and Styled
      Components
    </S.Subtitle>
    <S.Image
      src="/images/star.svg"
      alt="animation of a star rotating 360 degrees"
    />
  </S.Wrapper>
);

export default Main;
