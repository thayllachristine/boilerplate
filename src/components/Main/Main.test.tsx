import { render } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('should render correctly', () => {
    const { container } = render(<Main />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the right heading', () => {
    const { getByText } = render(<Main />);
    expect(getByText(/boilerplate/i)).toBeInTheDocument();
  });

  it('should render the right subtitle/description', () => {
    const { getByTestId } = render(<Main />);
    expect(getByTestId(/subtitle/i)).toBeInTheDocument();
  });

  it('should render image with alt text', () => {
    const { getByAltText } = render(<Main />);
    expect(getByAltText(/star/i)).toBeInTheDocument();
  });

  it('should render the colors correctly', () => {
    const { container, debug } = render(<Main />);
    debug();
    expect(container.firstChild).toHaveStyle({ 'background-color': '#333333' });
    expect(container.firstChild).toHaveStyle({ color: '#fff' });
  });
});
