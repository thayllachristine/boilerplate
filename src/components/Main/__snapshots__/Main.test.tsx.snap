// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #333333;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
  width: 100%;
}

.c1 {
  font-size: 2.5rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2rem;
  font-weight: 400;
  margin-bottom: 3rem;
}

.c3 {
  -webkit-animation: eoUyJr 10s linear infinite;
  animation: eoUyJr 10s linear infinite;
  height: min(30rem,100%);
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    boilerplate
  </h1>
  <h2
    class="c2"
    data-testid="subtitle"
  >
    A simple project starter to work with React, NextJS, TypeScript and Styled Components
  </h2>
  <img
    alt="animation of a star rotating 360 degrees"
    class="c3"
    src="/images/star.svg"
  />
</main>
`;
